app/src/main/java/com/example/habitsmasher/DaysTracker.java

```diff
@@ -103,13 +122,21 @@ public String getDays(){
 
 
     /**
-     * Takes a string with the days and converts them to days of the week.
-     * The string takes the format "day1 day2 ... day7", where each day is the first 2 letters of that day, separated by a space. The string can be in any order.
-     * EX) "MO FR WE TH"
+     * This method is the public selector for setting days. It can take either a string or an arraylist
+     * and convert the values within into the proper format for the tracker.
      *
-     * @param days The days to select
+     * @param days The days to select, either {@link String} or {@link ArrayList}
      */
-    public void setDays(String days){
+    public void setDays(Object days){
+        if (days instanceof String) {
+            setDaysFromString((String) days);
+        }
+        else if (days instanceof ArrayList){      // can't specify types in list unfortunately
+            setDaysFromList((ArrayList<String>) days);
+        }
+    }
```

Comment:
I split up the setDays method into two bits, to handle two different cases with one method. This makes it very convenient and scalable.